extensions/amp-story/0.1/amp-story.js

```diff
@@ -769,7 +784,9 @@ export class AmpStory extends AMP.BaseElement {
     this.updateBackground_(targetPage.element, /* initial */ !this.activePage_);
 
     // TODO(alanorozco): decouple this using NavigationState
-    this.systemLayer_.setActivePageIndex(pageIndex);
+    if (!targetPage.element.hasAttribute(ADVERTISEMENT_ATTR_NAME)) {
```

Comment:
maybe move the `hasAttribute` check to amp-story-page and expose it as `isAdvertisement()`?

Reply:
same thought about changing the pre-existing code as much as possible. If you think it's better I'm happy to change.

Reply:
moved to page logic